server/hotrod/src/main/java/org/infinispan/server/hotrod/CacheRequestProcessor.java

```diff
@@ -553,12 +551,13 @@ void addClientListener(HotRodHeader header, Subject subject, byte[] listenerId,
                           String filterFactory, List<byte[]> filterParams, String converterFactory,
                           List<byte[]> converterParams, boolean useRawData, int listenerInterests, int bloomBits) {
       Object span = RequestTracer.requestStart(HotRodOperation.ADD_CLIENT_LISTENER.name());
-      AdvancedCache<byte[], byte[]> cache = server.cache(server.getCacheInfo(header), header, subject);
+      HotRodCacheInfo cacheInfo = server.getCacheInfo(header);
+      AdvancedCache<byte[], byte[]> cache = server.cache(cacheInfo, header, subject);
       BloomFilter<byte[]> bloomFilter = null;
       if (bloomBits > 0) {
          bloomFilter = MurmurHash3BloomFilter.createConcurrentFilter(bloomBits);
-         BloomFilter<byte[]> priorFilter = bloomFilters.putIfAbsent(header.cacheName, bloomFilter);
-         assert priorFilter == null;
+         boolean filterAdded = cacheInfo.setBloomFilter(bloomFilter);
+         assert filterAdded;
```

Comment:
@wburns I kept the logic here, but it doesn't feel right to me: `addClientListener` will be called multiple times per cache, at least once for each near cache client connecting to this server.